pkg/artifactory/resource/security/resource_artifactory_scoped_token.go

```diff
@@ -0,0 +1,285 @@
+			SetPathParam("id", data.Id()).
+			SetResult(&accessToken).
+			Get(accessTokenUrl)
+
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		return packAccessToken(data, accessToken)
+	}
+
+	var packAccessTokenPostResponse = func(d *schema.ResourceData, accessToken AccessTokenPostResponse) diag.Diagnostics {
+		setValue := util.MkLens(d)
+
+		var errors []error
+		errors = append(errors, setValue("scopes", strings.Split(accessToken.Scope, " "))...)
+		errors = append(errors, setValue("expires_in", accessToken.ExpiresIn)...)
+		errors = append(errors, setValue("access_token", accessToken.AccessToken)...)
+		errors = append(errors, setValue("token_type", accessToken.TokenType)...)
+
+		if len(errors) > 0 {
+			return diag.Errorf("failed to pack access token from POST response %q", errors)
+		}
+
+		return nil
+	}
+
+	var accessTokenCreate = func(ctx context.Context, data *schema.ResourceData, m interface{}) diag.Diagnostics {
+		accessToken, err := unpackAccessTokenPostRequest(data)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		accessToken.GrantType = "client_credentials"
+
+		result := AccessTokenPostResponse{}
+		_, err = m.(*resty.Client).R().
+			SetBody(accessToken).
+			SetResult(&result).
+			Post(accessTokensUrl)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		data.SetId(result.Id())
+
+		diags := packAccessTokenPostResponse(data, result)
+		if diags != nil {
+			return diags
+		}
+
+		return accessTokenRead(ctx, data, m)
+	}
+
+	var accessTokenDelete = func(ctx context.Context, data *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+		_, err := m.(*resty.Client).R().
+			SetPathParam("id", data.Id()).
+			Delete(accessTokenUrl)
+
+		if err != nil {
+			d := &util.ResourceData{data}
+
+			expiry := time.Unix(int64(d.GetInt("expiry", false)), 0)
+			issuedAt := time.Unix(int64(d.GetInt("issued_at", false)), 0)
+
+			return diag.Diagnostics{{
+				Severity: diag.Error,
+				Summary:  fmt.Sprintf("Failed to revoke scoped token %s", data.Id()),
+				Detail:   fmt.Sprintf("Token expiration time: %s, issued at: %s", expiry, issuedAt),
+			}}
+		}
+
+		data.SetId("")
+
+		return nil
+	}
+
+	return &schema.Resource{
+		CreateContext: accessTokenCreate,
+		ReadContext:   accessTokenRead,
+		DeleteContext: accessTokenDelete,
+
+		Schema: map[string]*schema.Schema{
+			"username": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringLenBetween(1, 255)),
+				Description:      "The user name for which this token is created. The username is based on the authenticated user - either from the user of the authenticated token or based on the username (if basic auth was used). The username is then used to set the subject of the token: <service-id>/users/<username>. Limited to 255 characters.",
+			},
+			"scopes": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
```

Comment:
You have to have some kind of validation function

Reply:
We can use a regex to validate service ID which is in this format `jfrt@01e3symwq999mx00epqaw31qnm`